sphinx/ext/autodoc/__init__.py

```diff
@@ -1222,7 +1231,10 @@ class DocstringStripSignatureMixin(DocstringSignatureMixin):
     feature of stripping any function signature from the docstring.
     """
     def format_signature(self, **kwargs: Any) -> str:
-        if self.args is None and self.config.autodoc_docstring_signature:  # type: ignore
+        if (
+            self.args is None
+            and self.config.autodoc_docstring_signature  # type: ignore[attr-defined]
```

Comment:
```suggestion
            and self.config.autodoc_docstring_signature  # type: ignore[attr-defined] # NoQA: W503 
```

Reply:
is this still required? not seeing any lint errors after rebase

Reply:
The 'and' should start the line (per Knuth) -- I need to update the linting rules at some point.

A

Reply:
i can do it here if you like?